db/view/build_progress_virtual_reader.hh

```diff
@@ -72,22 +72,22 @@ class build_progress_virtual_reader {
                 tracing::trace_state_ptr trace_state,
                 streamed_mutation::forwarding fwd,
                 mutation_reader::forwarding fwd_mr)
-                : flat_mutation_reader::impl(std::move(legacy_schema), permit)
+                : flat_mutation_reader_v2::impl(std::move(legacy_schema), permit)
                 , _scylla_next_token_col(scylla_views_build_progress.schema()->get_column_definition("next_token")->id)
                 , _scylla_generation_number_col(scylla_views_build_progress.schema()->get_column_definition("generation_number")->id)
                 , _legacy_last_token_col(_schema->get_column_definition("last_token")->id)
                 , _legacy_generation_number_col(_schema->get_column_definition("generation_number")->id)
                 , _legacy_slice(slice)
                 , _slice(adjust_partition_slice())
-                , _underlying(scylla_views_build_progress.make_reader(
+                , _underlying(upgrade_to_v2(scylla_views_build_progress.make_reader(
```

Comment:
Table now has `make_reader_v2()`.

Reply:
Which is the native actually, so the above will cause a v2->v1->v2 conversion cycle.

Reply:
Will adjust.